client/lib/lib.rs

```diff
@@ -93,23 +93,35 @@ pub fn make_deploy(
     deploy_params: DeployStrParams<'_>,
     session_params: SessionStrParams<'_>,
     payment_params: PaymentStrParams<'_>,
+    force: bool,
 ) -> Result<()> {
-    let output = deploy::output_or_stdout(none_if_empty(maybe_output_path)).map_err(|error| {
-        Error::IoError {
-            context: format!(
-                "unable to get file or stdout, provided '{:?}'",
-                maybe_output_path
-            ),
-            error,
-        }
+    let maybe_output_path = none_if_empty(maybe_output_path);
+    let path_exists = if let Some(p) = maybe_output_path {
+        PathBuf::from(p).exists()
+    } else {
+        false
+    };
+
+    let output = deploy::output_or_stdout(maybe_output_path).map_err(|error| Error::IoError {
+        context: format!(
+            "unable to get file or stdout, provided '{:?}'",
+            maybe_output_path
+        ),
+        error,
     })?;
 
-    Deploy::with_payment_and_session(
-        deploy_params.try_into()?,
-        payment_params.try_into()?,
-        session_params.try_into()?,
-    )?
-    .write_deploy(output)
+    match (path_exists, force) {
+        (true, false) => {
+            let pb = PathBuf::from(maybe_output_path.unwrap());
+            Err(Error::FileAlreadyExists(pb))
+        }
+        (true, true) | (false, true) | (false, false) => Deploy::with_payment_and_session(
```

Comment:
Okay nitpick, but maybe mark these as constants, `const PATH_EXISTS: bool = true` , I don't think we should have naked booleans especially in this case.

Reply:
I don't understand how using a constant would be better than using an immutable `let` here, because `path_exists` could be true or false depending on if the path can be found.